File: src/gestorAplicacion/administracion/Salon.java
package gestorAplicacion.administracion;
import java.util.ArrayList;
import java.io.Serializable;

public class Salon implements Serializable{
        private String lugar;
        private int aforo;
        private Horario horario;
        public static ArrayList<Salon> salones= new ArrayList<Salon>();
        private static final long serialVersionUID = 1L;

        public Salon(String lugar,int aforo){
                this.lugar = lugar;
                this.aforo = aforo;
                this.horario = new Horario();
                salones.add(this);
        }
        
        public static String mostrarSalones() {
                String retorno = "";
                int i = 1;
                for(Salon salon:salones) {
                        retorno += (i++)+". "+salon.lugar+".\n";
                }
                return retorno;
        }

        public String getLugar() {
                return lugar;
        }

        public void setLugar(String lugar) {
                this.lugar = lugar;
        }

        public int getAforo() {
                return aforo;
        }

        public void setAforo(int aforo) {
                this.aforo = aforo;
        }

        public Horario getHorario() {
                return horario;
        }

        public void setHorario(Horario horario) {
                this.horario = horario;
        }

        public static ArrayList<Salon> getSalones() {
                return salones;
        }

        public static void setSalones(ArrayList<Salon> salones) {
                Salon.salones = salones;
        }
    
}

File: Python/src/gestorAplicacion/administracion/Salon.py
from gestorAplicacion.administracion.Horario import Horario


class Salon:
    _salones = []

    def __init__(self, lugar, aforo):
        self._lugar = lugar
        self._aforo = aforo
        self._horario = Horario()  # no estoy segura de como hacer eso aca
        Salon._salones.append(self)

    # METODOS

    @classmethod
    def mostrarSalones(cls):
        retorno = ""
        i = 1
        for salon in Salon._salones:
            i += 1
            retorno += str(i) + ". " + salon._lugar + ".\n"
        return retorno
    
    @classmethod
    def nombresSalones(cls):
        nombres = []
        for salon in Salon._salones:
            nombres.append(salon.getLugar())
        return nombres
    
    @classmethod
    def encontrarSalon(cls, salon):
        for s in Salon._salones:
            if s.getLugar()==salon:
                return s

    # Setters y Getters

    def getLugar(self):
        return self._lugar

    def setLugar(self, lugar):
        self._lugar = lugar

    def getAforo(self):
        return self._aforo

    def setLugar(self, aforo):
        self._aforo = aforo

    def getLugar(self):
        return self._lugar

    def setLugar(self, lugar):
        self._lugar = lugar

    def getHorario(self):
        return self._horario

    def setHorario(self, horario):
        self._horario = horario

    @classmethod
    def getSalones(cls):
        return cls._salones

    @classmethod
    def setSalones(cls, salones):
        cls._salones = salones
